app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt

```diff
@@ -1,31 +1,155 @@
 package com.android.voyageur.ui.profile
 
-import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.Scaffold
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccountCircle
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.android.voyageur.model.user.User
+import com.android.voyageur.model.user.UserViewModel
 import com.android.voyageur.ui.navigation.BottomNavigationMenu
 import com.android.voyageur.ui.navigation.LIST_TOP_LEVEL_DESTINATION
 import com.android.voyageur.ui.navigation.NavigationActions
+import com.android.voyageur.ui.navigation.Route
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.delay
 
 @Composable
-fun ProfileScreen(
-    navigationActions: NavigationActions,
-) {
-
-  Scaffold(
-      modifier = Modifier.testTag("profileScreen"),
-      bottomBar = {
-        BottomNavigationMenu(
-            onTabSelect = { route -> navigationActions.navigateTo(route) },
-            tabList = LIST_TOP_LEVEL_DESTINATION,
-            selectedItem = navigationActions.currentRoute())
-      },
-      content = { pd ->
-        Text(
-            modifier = Modifier.padding(pd).testTag("emptyProfilePrompt"),
-            text = "You can't customize your profile yet.")
-      })
+fun ProfileScreen(userViewModel: UserViewModel, navigationActions: NavigationActions) {
+    // Observing user and loading state from the UserViewModel
+    val user by userViewModel.user.collectAsState()
+    val isLoading by userViewModel.isLoading.collectAsState()
+
+    // Firebase current user
+    val currentUser = FirebaseAuth.getInstance().currentUser
+
+    var isSigningOut by remember { mutableStateOf(false) }
+
+    // Handle sign-out with navigation in a LaunchedEffect
+    if (isSigningOut) {
+        LaunchedEffect(isSigningOut) {
+            delay(300)
+            navigationActions.navigateTo(Route.AUTH)
+        }
```

Comment:
Good work! Quick question: here shouldn't you also call `userViewModel.signOutUser()` ? For me when I test manually, once I log out, I'm redirected to the Sign In screen, but then my user is remembered and I'm signed in directly once I press the button.

Reply:
You're absolutely right. The issue you're encountering happens because the FirebaseAuth.signOut() method is likely not being called explicitly in the ProfileScreen, so Firebase remembers the user session when you try to log back in.